client/driver/docker.go

```diff
@@ -249,8 +249,19 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 		exposedPorts := map[docker.Port]struct{}{}
 
 		for _, port := range network.ReservedPorts {
+			// By default we will map the allocated port 1:1 to the container
+			containerPortInt := port.Value
+
+			// If the user has mapped a port using port_map we'll change it here
+			if len(driverConfig.PortMap) == 1 {
```

Comment:
the first block is always 0. If we're worried about multiple port_map entries we should normalize this instead.